react_projects/rick-and-morty-web/src/components/tableContent/TableContent.js

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import RawItem from '../rawItem/RawItem';
+
+import './TableContent.css'
+import Pagination from '../pagination/Pagination';
+import FilterBar from '../filterBar/filterBar';
+import Loader from '../Loader/Loader';
+import useDebounce from '../../hooks/useDebounce';
+
+const TableContent = () => {
+
+  const [data, setData] = useState(null);
+  const [searchInput, setSearchInput] = useState('')
+  const debouncedSearchValue = useDebounce(searchInput)
+  const [pageNumber, setPageNumber] = useState(1);
+
+  useEffect(() => {
+    fetch(`https://rickandmortyapi.com/api/character?page=${pageNumber}&name=${debouncedSearchValue}`)
+      .then(response => response.json())
+      .then(json => setData(json))
+      .catch(error => {setData(null);console.error(error)});
+  }, [pageNumber, debouncedSearchValue]);
+
+  const CharacterList = () => {
+    return(
+      <>
+        {data.results.map((item) => {
+          return <RawItem character={item}/>
+        })}
+      </>
+    )
+  }
+  return (
+    <div className='table-main-container'>
+      {data ?
+        <>
+          <FilterBar searchInput={searchInput} updateSearch={setSearchInput}></FilterBar>
+          {data.info ?
+            <>
+              <Pagination actualPage={pageNumber} totalPages={data?.info?.pages} paginate={setPageNumber}/>
```

Comment:
```suggestion
              <Pagination actualPage={pageNumber} totalPages={data.info.pages} paginate={setPageNumber}/>
```

at this point, you are sure `data.info` is present since you checked it in the line 38